Std/Data/List/SplitOnList.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2024 Bulhwi Cha, Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bulhwi Cha, Mario Carneiro
+-/
+import Std.Data.Nat.Lemmas
+import Std.Data.List.Lemmas
+
+namespace List
+
+/-- Returns `(l₁, l₂)` for the first split `l = l₁ ++ l₂` such that `P l₂` returns true. -/
+def splitOnceP (P : List α → Bool) (l : List α) : Option (List α × List α) :=
```

Comment:
If I only read the name and signature (and not the doc-string), I would be surprised by the behaviour here; I would have guessed "`P l₁` returns true".

I appreciate that this is motivated by `splitOnList`, but could we either have a more information name, or make this private? (or defined via a `where` clause in `splitOnList`?)

Reply:
Do you think `splitOnceRightP` is a good name?

Reply:
Sounds better to me.